### DISCUSSION TITLE
Status API does not return metadata added in .manifest file of the OPA bundle.

### DISCUSSION THREAD

**Discussion by apatade-tl** [open-policy-agent/community#613] [OPA and Rego]
Hi team,

I am trying to add metadata via the .manifest file of the bundle along with other files like data.json, evaluator.rego and policy rego files.

Only revision is returned as the active_revision from the .manifest file. **Is there any configuration I am missing for which the metadata content from the .manifest file is not getting returned?**

Command used to run the server:

> opa run --server --set bundles.example.resource=bundle.tar.gz --set services.example.url=http://localhost:8080/ --set status.service=example

.manifest file containing metadata:
```
{
    "revision": "v1.0.0",
    "metadata": {
        "environment": "production",
        "timestamp": "2024-07-12T12:34:56Z"
    }
}
```

Status API returns:
```
{
    "labels":
    {
        "id": "b0466ccb-5976-4a09-a755-667f7da44e4f",
        "version": "0.66.0"
    },
    "bundles":
    {
        "example":
        {
            "name": "example",
            "active_revision": "v1.0.0",
            "last_successful_activation": "2024-07-13T13:29:18.351839Z",
            "type": "snapshot",
            "size": 5258,
            "last_successful_download": "2024-07-13T13:29:18.328496Z",
            "last_successful_request": "2024-07-13T13:29:18.328496Z",
            "last_request": "2024-07-13T13:29:18.328496Z",
            "metrics":
            {
                "timer_bundle_request_ns": 189486000,
                "timer_rego_load_bundles_ns": 34545583,
                "timer_rego_module_compile_ns": 22106083,
                "timer_rego_module_parse_ns": 9814789
            }
        }
    },
    "metrics":
    {...},
    "plugins":
    {
        "bundle":
        {
            "state": "OK"
        },
        "discovery":
        {
            "state": "OK"
        },
        "status":
        {
            "state": "OK"
        }
    }
}
```

**Comment by charlieegan3** [comment]
Hi @apatade-tl, Have you tried accessing the bundle metadata via `data.system` instead? This information isn't available in the status API.

**Reply by apatade-tl** [reply]
@charlieegan3 Yes, getting the metadata in the data.system API. 

Use case:
We have a single pod that is responsible for the authorization and listens to the `status` and `bundle` load API's of OPA containers/servers of multiple (100s) pods (specific to multiple applications). We do not want to jump into the logic of the address resolutions of OPA containers from the different application's pods to hit `/v1/data/system` API.

Instead/hence, we are looking for any way to get the metadata (any custom data) via status API, as like revision is automatically returned as active_revision.

**Reply by charlieegan3** [reply]
How are the apps able to get the status API if they can't get the `/v1/data/system` API?

**Reply by apatade-tl** [reply]
Apps do not get the status API, but each OPA server is configured with status.service property to use/hit single authorization pod's endpoint.
As we have used config `--set status.service=example` hence the example service's `/status` (http://localhost:8080/status) would be hit which would be automatic whenever the state of the bundle changes inside the OPA server.

**Reply by charlieegan3** [reply]
Ahh, sorry. This makes sense to me now.

I think the best course of action here is to open a new issue for this one. You can do [so here](https://github.com/open-policy-agent/opa/issues/new).

I think that adding a new field for metadata to the Status struct should be fairly easy to agree on. But in the issue, make sure to give the details about the use case and make a case for the value this gives you too!

(https://github.com/open-policy-agent/opa/blob/cb427253ac82b501bafa1fc784f45b7b7377c66d/plugins/bundle/status.go#L24)

**Reply by apatade-tl** [reply]
There are ways to do it via custom plugin and so, but I feel the use case stands very valid and if something of this sort is coming with OPA itself, then people/community would get benefit from this too. As to have the functionality of returning metadata (or any custom data for that matter) which is present in the bundle as part of status API. Or any such mechanism for the purpose.

Right, had opened one this morning at https://github.com/open-policy-agent/opa/issues/6871.

Sure, I would add more details about the use case and make a proposal. Thanks for looking in! :)

**Reply by charlieegan3** [reply]
Ahh, I see it now, thanks for opening the issue. I think it'd be good to have a little more detail about the value this would add for you.

I've relabelled it as a feature request since it's not a bug but rather something new to add.
